src/main/kotlin/com/getkeepsafe/dexcount/CountReporter.kt

```diff
@@ -0,0 +1,184 @@
+
+            withStyledOutput { out ->
+                out.println("Dexcount name:    $projectName")
+                out.println("Dexcount version: $projectVersion")
+                out.println("Dexcount input:   $inputRepresentation")
+            }
+        }
+    }
+
+    private fun printSummary() {
+        if (isInstantRun) {
+            withStyledOutput(Color.RED) { out ->
+                out.println("Warning: Instant Run build detected!  Instant Run does not run Proguard; method counts may be inaccurate.")
+            }
+        }
+
+        val color = if (packageTree.methodCount < 50000) Color.GREEN else Color.YELLOW
+
+        withStyledOutput(color) { out ->
+            fun percentUsed(count: Int): String {
+                val used = (count.toDouble() / MAX_DEX_REFS) * 100.0
+                return String.format("%.2f", used)
+            }
+
+            val percentMethodsUsed = percentUsed(packageTree.methodCount)
+            val percentFieldsUsed = percentUsed(packageTree.fieldCount)
+            val percentClassesUsed = percentUsed(packageTree.classCount)
+
+            val methodsRemaining = max(MAX_DEX_REFS - packageTree.methodCount, 0)
+            val fieldsRemaining = max(MAX_DEX_REFS - packageTree.fieldCount, 0)
+            val classesRemaining = max(MAX_DEX_REFS - packageTree.classCount, 0)
+
+            val (methodCount, fieldCount, classCount) = when {
+                isAndroidProject -> listOf(packageTree.methodCount, packageTree.fieldCount, packageTree.classCount)
+                else -> listOf(packageTree.methodCountDeclared, packageTree.fieldCountDeclared, packageTree.classCountDeclared)
+            }
+
+            out.println("Total methods in $inputRepresentation: $methodCount ($percentMethodsUsed% used)")
+            out.println("Total fields in $inputRepresentation:  $fieldCount ($percentFieldsUsed% used)")
+            out.println("Total classes in $inputRepresentation:  $classCount ($percentClassesUsed% used)")
+
+            if (isAndroidProject) {
+                out.println("Methods remaining in $inputRepresentation: $methodsRemaining")
+                out.println("Fields remaining in $inputRepresentation:  $fieldsRemaining")
+                out.println("Classes remaining in $inputRepresentation:  $classesRemaining")
+            }
+        }
+
+        summaryFile.parentFile.mkdirs()
+        summaryFile.createNewFile()
+
+        val headers = "methods,fields,classes"
+        val counts = "${packageTree.methodCount},${packageTree.fieldCount},${packageTree.classCount}"
+
+        summaryFile.printWriter().use { writer ->
+            writer.println(headers)
+            writer.println(counts)
+        }
+
+        if (options.teamCityIntegration || config.teamCitySlug != null) {
+            withStyledOutput { out ->
+                val slug = "Dexcount" + (config.teamCitySlug?.let { "_" + it.replace(' ', '_') } ?: "")
```

Comment:
`?: ""` can be `.orEmpty()` ?

Reply:
I just copy/pasted this from the task implementation, but I agree with you here that this line of code should be cleaned up.  Even better, it should be _broken_ up IMO.